src/metabase/feature_extraction/feature_extractors.clj

```diff
@@ -459,29 +463,47 @@
   (redux/post-complete
    (redux/fuse {:histogram (redux/pre-step
                             h/histogram
-                            (stats/somef (comp count u/jdbc-clob->str)))})
+                            (somef (comp count u/jdbc-clob->str)))})
    (merge-juxt
     (field-metadata-extractor field)
     histogram-extractor)))
 
-(defn- quarter
-  [dt]
-  (-> dt t/month (/ 3) Math/ceil long))
+(defprotocol Quarter
+  "Quarter-of-year functionality"
+  (quarter [dt] "Return which quarter (1-4) given date-like object falls into."))
+
+(extend-type java.util.Date
+  Quarter
+  (quarter [dt]
+    (-> dt .getMonth inc (* 0.33) Math/ceil long)))
+
+(extend-type org.joda.time.DateTime
+  Quarter
+  (quarter [dt]
+    (-> dt t/month (* 0.33) Math/ceil long)))
 
 (defmethod feature-extractor DateTime
-  [_ field]
+  [_ {:keys [base_type unit] :as field}]
   (redux/post-complete
-   (redux/pre-step
-    (redux/fuse {:histogram         (redux/pre-step h/histogram t.coerce/to-long)
-                 :histogram-hour    (redux/pre-step h/histogram-categorical
-                                                    (stats/somef t/hour))
-                 :histogram-day     (redux/pre-step h/histogram-categorical
-                                                    (stats/somef t/day-of-week))
-                 :histogram-month   (redux/pre-step h/histogram-categorical
-                                                    (stats/somef t/month))
-                 :histogram-quarter (redux/pre-step h/histogram-categorical
-                                                    (stats/somef quarter))})
-    t.format/parse)
+   (redux/fuse (merge
+                {:histogram         (redux/pre-step
+                                     h/histogram
+                                     (somef (memfn ^java.util.Date getTime)))
+                 :histogram-day     (redux/pre-step
+                                     h/histogram-categorical
+                                     (somef (memfn ^java.util.Date getDay)))
+                 :histogram-month   (redux/pre-step
+                                     h/histogram-categorical
+                                     #(when %
+                                        (inc (.getMonth ^java.util.Date %))))
+                 :histogram-quarter (redux/pre-step
+                                     h/histogram-categorical
+                                     (somef quarter))}
+                (when-not (or (isa? base_type :type/Date)
+                              (#{:day :month :year :quarter :week} unit))
+                  {:histogram-hour (redux/pre-step
+                                    h/histogram-categorical
+                                    (somef (memfn ^java.util.Date getHours)))})))
```

Comment:
All of these `.getXXXX()` calls on Date are deprecated, I think they want you to use Calendar instead. I don't think it needs to hold up getting merged as I doubt the methods are going anywhere ,but something we should probably fix up.